scenes/Player/Player.gd

```diff
@@ -35,22 +38,23 @@ func _physics_process(delta: float) -> void:
 	
 	if Input.is_action_pressed("rotate_right"):
 		rotation_degrees+= rotation_speed * delta
-		
-	velocity = move_and_slide(velocity, Vector2(0, 0), false, 4, 0.785, false)
-	for i in get_slide_count():
-		var collision = get_slide_collision(i)
-		print("Collided with ", collision.collider.name)
-		if collision.collider.name.begins_with("Asteroid"):
-			collision.collider.apply_central_impulse(-collision.normal * inertia)
-			if PlayerState.hull_health > 0:
-				PlayerState.hull_health = PlayerState.hull_health -10
-			else:
-				print("health is 0")
-				visible = false
-
+	
+	_get_collision_damage()
```

Comment:
```suggestion
	_apply_collision_damage()
```

Suggestion: `apply` makes more sense to me since usually when I see `get` in a function name I'm expecting it to return something.

Reply:
I believe it's registering more than one collision. One solution is to make the player immune to damage after taking damage for a short period, it's common in games and player friendly. In general, I think I would prefer an "area entered" signal or the like for tracking damage. We can conceptualize an asteroid as a flying hurt box. Apply is a good change.